cmd/kubectl-k8ssandra/register/command.go

```diff
@@ -0,0 +1,72 @@
+package register
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+var RegisterClusterCmd = &cobra.Command{
+	Use:   "register [flags]",
+	Short: "register a data plane into the control plane.",
+	Long:  `register creates a ServiceAccount on a source cluster, copies its credentials and then creates a secret containing them on the destination cluster. It then also creates a ClientConfig on the destination cluster to reference the secret.`,
+	Run:   entrypoint,
+}
+
+func Init(cmd *cobra.Command, streams genericclioptions.IOStreams) {
+	RegisterClusterCmd.Flags().String("source-kubeconfig",
+		"",
+		"path to source cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("dest-kubeconfig",
+		"",
+		"path to destination cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("source-context", "", "context name for source cluster")
+	RegisterClusterCmd.Flags().String("dest-context", "", "context name for destination cluster")
+	RegisterClusterCmd.Flags().String("source-namespace", "k8ssandra-operator", "namespace containing service account for source cluster")
+	RegisterClusterCmd.Flags().String("dest-namespace", "k8ssandra-operator", "namespace where secret and clientConfig will be created on destination cluster")
+	RegisterClusterCmd.Flags().String("serviceaccount-name", "k8ssandra-operator", "serviceaccount name for destination cluster")
+	RegisterClusterCmd.Flags().String("destination-name", "remote-k8ssandra-operator", "name for remote clientConfig and secret on destination cluster")
+
+	if err := RegisterClusterCmd.MarkFlagRequired("source-context"); err != nil {
+		panic(err)
```

Comment:
No panics, handle errors nicely and provide users with feedback.

Reply:
I'm not sure that this is actually an issue:

1. This is a fatal error, there is no possible recovery.
2. This is an internal error, no user action can prevent it so surfacing it to the user is pointless.
3. `Init` (like the Cobra `AddCommand` function) does not permit returning an error. So I'm not sure what a "nicer" handling would be here.
4. I can't see any other examples in this repo where you've actually added validation to ensure that certain flags are provided. If there was another way to do it, it is unclear to me.

I'm not just calling `panic()` without cause here, I have actually considered whether errors are recoverable or something the user can affect.

Reply:
No, ``panic()`` is indicative of a fatal error as in a bug. Not a user facing error, where instead we simply return the help screen to the user telling them you're missing a flag and then show them the list which flags are available.

This throws the user the same as unchecked NullPointerException. It looks like a bug, looks ugly and is not helpful to the user. 

Look at my ls command with invalid parameters:

```console
➜  ~ ls -l34jh
ls: invalid option -- '3'
Try 'ls --help' for more information.
➜  ~
```

It doesn't give me ``Segmentation fault``and a stacktrace like panic() here does, no one would think that's acceptable. This isn't unrecoverable error from the point of view of the program, you simply have missing parameter - tell the user that. That's one of the input / outputs of a command line program. Handle errors gracefully always, especially when it's easy is Cobra. 

Cobra has the ability to show the correct parameters when you exit with an error. Use that feature.

Reply:
I think we disagree on the nature of this function call. My understanding is that this returns an error when the flag does not exist on the actual Cmd instance - `return fmt.Errorf("no such flag -%v", name)`.

You seem to believe that this error will be hit if the user does not supply the required flag. I don't think your understanding of this failure is correct. It is internal, a bug, and not recoverable. Therefore it is a panic.